src/replication.c

```diff
@@ -831,34 +1127,93 @@ ask_replica_status_all(spec_t *spec)
 	MTX_UNLOCK(&spec->rq_mtx);
 }
 
-/*
- * This function processes the status response received from replica
- * and update spec's healthy/degrade count
- */
+static void
+handle_prepare_for_rebuild_resp(spec_t *spec, zvol_io_hdr_t *hdr,
+    mgmt_ack_t *ack_data, mgmt_cmd_t *mgmt_cmd)
+{
+
+	int ret = 0;
+	rcommon_mgmt_cmd_t *rcomm_mgmt = mgmt_cmd->rcomm_mgmt;
+	mgmt_ack_t *buf = (mgmt_ack_t *)rcomm_mgmt->buf;
+	
+	if (hdr->status != ZVOL_OP_STATUS_OK) {
+		rcomm_mgmt->cmds_failed++;
+	} else {
+		memcpy(&buf[rcomm_mgmt->cmds_succeeded++], ack_data,
+		    sizeof (mgmt_ack_t));
+	}
+
+	if (rcomm_mgmt->cmds_sent == rcomm_mgmt->cmds_succeeded) {
+		ret = start_rebuild(buf, spec->target_replica, rcomm_mgmt->buf_size);
+		if (ret == 0) {
+			rcomm_mgmt->buf = NULL;
+			REPLICA_LOG("Rebuild triggered on Replica ip:%s port:%d"
+			    " state:%d\n", spec->target_replica->ip,
+			    spec->target_replica->port, spec->target_replica->state);
+		} else {
+			MTX_LOCK(&spec->rq_mtx);
```

Comment:
we won't be needing this lock..

Reply:
All over the places, I am updating this "spec" struct under this lock.  Removing it from this place will not be consistent with coding. 

There could be a separate discussion that why I am using lock to update this struct. We can discuss that and decide to remove lock from all the place where I am updating spec.

Reply:
actually, we are not using lock to update these variables at any place other than here. We are safe to remove lock here also.